src/redistributors/morpho/Distributor.sol

```diff
@@ -0,0 +1,118 @@
+// SPDX-License-Identifier: BUSL-1.1
+
+pragma solidity 0.8.23;
+
+import { ERC20Helper } from "../../../lib/erc20-helper/src/ERC20Helper.sol";
+
+import { IMorphoVaultFactoryLike, IMorphoVaultLike, IWrappedMLike } from "./Dependencies.sol";
+
+import { AccountFundsDistribution } from "./AccountFundsDistribution.sol";
+import { VaultFundsDistribution } from "./VaultFundsDistribution.sol";
+
+/**
+ * @title Morpho Distributor (Dual-Cascade, Cumulative Distribution)
+ * @dev   A contract for distributing Morpho yield across accounts within vaults.
+ * @dev   Each distribution cascade is an ERC2222 model relying on tracking per-account "corrections".
+ */
+contract Distributor is AccountFundsDistribution, VaultFundsDistribution {
+    error InvalidVault();
+    error TransferFailed();
+    error TransferFromFailed();
+
+    address public immutable wrappedMToken;
+    address public immutable morphoBlue;
+    address public immutable morphoVaultFactory;
+
+    uint256 internal _lastWrappedMBalance;
+
+    mapping(address vault => uint256 lastCumulativeDistribution) internal _lastCumulativeDistributions;
+    mapping(address vault => mapping(address account => uint256 claimed)) internal _claims;
+
+    constructor(address wrappedMToken_, address morphoBlue_, address morphoVaultFactory_) {
+        wrappedMToken = wrappedMToken_;
+        morphoBlue = morphoBlue_;
+        morphoVaultFactory = morphoVaultFactory_;
+    }
+
+    /// @notice Allow `msg.sender` to deposit an amount of Morpho Vault shares to get access to WrappedM yield.
+    function deposit(address vault_, uint256 vaultShares_) external {
+        _revertIfNotMorphoVault(vault_);
+
+        if (!ERC20Helper.transferFrom(vault_, msg.sender, address(this), vaultShares_)) revert TransferFromFailed();
+
+        _addShares(_updateLastWrappedMBalance(), vault_, vaultShares_);
```

Comment:
Names and logic are not well described and clear to the reader. It can be improved by renaming methods, adding explicit comments, and/or splitting code into logical blocks.

There are 2 implicit distributions happening within `addShares`, 2 updates of distribution tracking variables, calculation of distributable variables, etc. 

Additionally structure of code and naming leads to VS code editor not navigating correctly which complicates comprehension and audit of this code piece.

Reply:
Updated to prefix all internal calls with `AccountFundsDistribution` or `VaultFundsDistribution`, and added comments.